flang/lib/Lower/OpenMP.cpp

```diff
@@ -487,15 +495,54 @@ void DataSharingProcessor::collectDefaultSymbols() {
   }
 }
 
-void DataSharingProcessor::privatize() {
+void DataSharingProcessor::privatize(
+    llvm::SetVector<mlir::omp::PrivateClauseOp> *privateInitializers) {
```

Comment:
For now, we do nothing with `privateInitializaers`. However, the plan is to populate this list and use it to link the created ops with the corresponding block arguments for the private variables.

Reply:
Thanks for taking a look!

Just to be sure we are on the same page, are you referring to using the push/pop scope mechanism instead of the `privatizerSymbolMap` I create in line 529 below?

The `privateInitializers` argument is not related to scoping and symbol map management. Its purpose is to be populated with the created ops so that we can use them as operands to the `omp.parallel`/`omp.target` ops (or any op with a `private` clause.

Reply:
The 2nd commit in the PR uses the push/pop scope mechanism. Maybe adding an getter to the symbol map is not the best approach (instead we might want to pass down the `Fortran::lower::SymMap &symTable` parameter from `genOMP`.